rover/src/main/kotlin/io/rover/rover/ui/BlockViewModel.kt

```diff
@@ -0,0 +1,133 @@
+package io.rover.rover.ui
+
+import android.graphics.Rect
+import io.rover.rover.core.domain.Block
+import io.rover.rover.core.domain.HorizontalAlignment
+import io.rover.rover.core.domain.Position
+import io.rover.rover.core.domain.VerticalAlignment
+
+open class BlockViewModel(
+    private val block: Block
+): BlockViewModelInterface {
+    override fun stackedHeight(bounds: Rect): Float = when(block.position) {
+        Position.Floating -> 0.0f
+        Position.Stacked -> {
+            // TODO: what about if the block top and bottom values are proportional (percentage?)
```

Comment:
You can only have vertical percentage based properties when the row has a fixed height. This will be enforced in the UI.